### ISSUE TITLE
Provide valid, concise, and meaningful alternative text for image buttons

### ISSUE THREAD

**Issue by blnkt** [lsst-epo/rubin-obs-client#50]
## Description
All image buttons should provide meaningful alternative text, contain no unnecessary titles, and hide redundant or unnecessary information from screen readers.

### Header/Footer

- [x] Search and Menu icon buttons should provide meaningful alternative text.
- [x] Search and Menu buttons' icon svgs should not have unnecessary title and should be hidden from screen reader using `aria-hidden="true"` or `role="presentation"`

![fd8a4d282b2875372396718d1ec9b77a6f7d2bae-1641232657](https://user-images.githubusercontent.com/8166060/152439459-5a71e2d3-2ec4-4eae-9fb1-c3803f530ba0.png)

### SlideBlock

- [x] The buttons that allow the user to select slides should have meaningful and descriptive names (e.g. "1,2,3,4,5"). This can be achieved by providing any of the following:
  - an alt attribute for input type image
  - an aria-label for nested icon font elements
  - an alt attribute for nested <img> elements
  - a referenced <title> element for svg images (the svg element should provide an aria-labelledby attribute that maps to the id attribute value of the nested title)

![4b511f45114f89d00b6cf674d75480bb17261766-1641297793](https://user-images.githubusercontent.com/8166060/152439524-c0b732c5-b9d5-4f36-9f1b-d228a83f6ae6.png)

#### Compliant Code Example:
```<ul class="slick-dots" style="display: block;"><li class=""><button>Go to slide1</button></li><li class="slick-active"><button> Go to slide 2</button></li><li class=""><button> Go to slide 3</button></li><li class=""><button>Go to slide 4</button></li><li class=""><button>Go to slide 5</button></li></ul>```

**Comment by dananjohnson** [comment]
Seems like we could only do one of these two solutions but not both:

> - [ ]  Search and Menu icon buttons should provide meaningful alternative text.
> - [ ] Search and Menu buttons' icon svgs should not have unnecessary title and should be hidden from screen reader using aria-hidden="true" or role="presentation"

I'm inclined to do the latter, since the icons are purely decorative, and to just make sure that the buttons have discernable text.

**Comment by dananjohnson** [comment]
It looks like the internals of the SlideBlock content element come from a third-party library called `react-slick`, so we don't have control over the markup. Given that LevelAccess flagged a number of issues with this thing, I think we'll need to find a replacement library.

**Comment by blnkt** [comment]
Well, for this issue I believe the slick slide api allows for [custom dots markup](https://react-slick.neostack.com/docs/api#appendDots) and [other controls](https://react-slick.neostack.com/docs/api#customPaging).  But, if there enough other SlideBlock related accessibility changes, then I leave it up to y'all if the best way to address overall is to jump to a different lib.

**Comment by dananjohnson** [comment]
Thanks for flagging those, @blnkt! I'm new to slick. It'd be great if we could solve all the a11y issues without finding a new library, so I'll review everything and assess.

**Comment by dananjohnson** [comment]
I ultimately ended up replacing `react-slick`. Issue [63](https://github.com/lsst-epo/rubin-obs-client/issues/63) was kinda the deal-breaker. I replaced it with a hook that loads up our old friend `flickity`, and a handful of components to render the slider, buttons, pagination, and live-region state announcements. I followed the [WAI-ARIA carousel design pattern](https://www.w3.org/TR/wai-aria-practices-1.1/examples/carousel/carousel-1.html) to get the a11y bits right.
